### PR TITLE
Expose readFile encoding option through vfs.src

### PR DESCRIPTION
**Pull request by limdauto**
Here is a PR for `src` regarding #23.

I will add support for `dest` if this is up to par :)

### CODE REVIEW

File: lib/src/getContents/index.js [hunk 1]
```diff
@@ -14,7 +15,9 @@ function getContents(opt) {
 
     // read and pass full contents
     if (opt.buffer !== false) {
-      return bufferFile(file, cb);
+      // set encoding type if user specifies it
+      var options = 'encoding' in opt ? {'encoding': opt.encoding} : null;
```
Comment (callumacrae) [inline lib/src/getContents/index.js]: Why not just `var options = { encoding: opt.encoding }`? If `opt.encoding` is undefined, it'll be passed through as undefined, which is fine.

**Comment by callumacrae** [comment]
lgtm except for the one thing

File: lib/src/getContents/index.js (hunk 1 above)
Comment (limdauto) [inline lib/src/getContents/index.js]: Thanks. Fixed.

**Comment by contra** [comment]
Test fail?

**Comment by limdauto** [comment]
@contra it's failing on master

File: lib/src/getContents/bufferFile.js [hunk 2]
```diff
@@ -3,12 +3,18 @@
...
-  fs.readFile(file.path, function(err, data) {
+function bufferFile(file, opt, cb) {
+  fs.readFile(file.path, opt, function(err, data) {
     if (err) {
       return cb(err);
     }
-    file.contents = stripBom(data);
+    // if user specifies encoding type, a string is returned
+    // so we need to convert it back to buffer
+    if (!Buffer.isBuffer(data) && typeof data === 'string') {
```
Comment (contra) [inline lib/src/getContents/bufferFile.js]: just typeof is sufficient here, no need for both checks. then you can make this a ternary

File: lib/src/getContents/index.js [hunk 3]
```diff
@@ -14,7 +15,9 @@ function getContents(opt) {
 
     // read and pass full contents
     if (opt.buffer !== false) {
-      return bufferFile(file, cb);
+      // set encoding type if user specifies it
+      var options = {'encoding': opt.encoding};
```
Comment (contra) [inline lib/src/getContents/index.js]: objects are multi-line, also encoding doesnt need quotes around it

File: lib/src/getContents/index.js [hunk 4]
```diff
@@ -14,7 +15,9 @@ function getContents(opt) {
...
+      return bufferFile(file, options, cb);
```
Comment (contra) [inline lib/src/getContents/index.js]: why not just pass the whole options object through?

**Comment by contra** [comment]
How does this work with writing files out? Aren't we going to need to specify the encoding there as well

File: lib/src/getContents/index.js (hunk 4 above)
Comment (limdauto) [inline lib/src/getContents/index.js]: I was trying to be explicit with the options so it matches readFile API.

File: lib/src/getContents/bufferFile.js (hunk 2 above)
Comment (limdauto) [inline lib/src/getContents/bufferFile.js]: That's true

**Comment by limdauto** [comment]
@contra I understand without passing encoding to writing out, all this is kind of pointless. I just made the PR to see if I'm on the right track. Will add `dest` shortly.

File: lib/src/getContents/bufferFile.js (hunk 2 above)
Comment (contra) [inline lib/src/getContents/bufferFile.js]:
```js
if (typeof data === 'string) {
  data = new Buffer(data, opt.encoding);
}
file.contents = stripBom(data);
```

would be cleanest imo

File: lib/src/getContents/index.js (hunk 4 above)
Comment (contra) [inline lib/src/getContents/index.js]: @limdauto passing the whole thing down is what is done for everything, makes it easier to add new option behavior later on down the road

**Comment by sindresorhus** [comment]
What use-case is this solving exactly? When would you ever need to be able to specify `ascii` encoding?

**Comment by phated** [comment]
Encoding doesn't really make sense because we always read in as buffer, which doesn't have an encoding.  Plugins use `toString()` on that buffer to turn it into a string, so there's not a good way to expose the encoding information without breaking all plugins.

**Comment by sindresorhus** [comment]
I would think https://github.com/heldinz/gulp-convert-encoding would be a better solution if you're using obscure encodings.

**Comment by limdauto** [comment]
Hi guys, so sorry I completely forgot about this PR. It was more than a month ago and I have indeed found out that it's better to convert my stuff to sane encoding first rather than requiring it from gulp/vinyl-fs. Thanks for your comments. I'm closing this.

**Pull request closed without merging**